src/simulation.hpp

```diff
@@ -703,6 +708,14 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
 			WritePlotFile();
 		}
 
+		// Writing Plot files at time intervals
+		double next_plot_file_time = next_plot_file_num * plotTimeInterval_;
+		if (plotTimeInterval_ > 0 && ((next_plot_file_time > cur_time) && (next_plot_file_time < (cur_time + dt_[0])))) {
```

Comment:
I think there might be a floating-point roundoff issue here. Is the last condition necessary?

Reply:
Do you mean- `(next_plot_file_time < (cur_time + dt_[0])`? Yes, we need to make sure that `next_plot_file_time` is within dt of `cut_time`. Because if `next_plot_file_time` is say cur_time + 3dt, then it would write 3 files instead of 1.

Reply:
Could you change the latter two conditions to just `next_plot_file_time <= cur_time`?